File: backend/app/schemas/transit.py
"""Transit API schemas (WEB-MAP-02)."""

from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

TransitMode = Literal["scheduled", "realtime"]


class TransitDepartureOut(BaseModel):
    route_short_name: str
    route_type: str
    headsign: str
    scheduled_at: datetime
    minutes: int
    realtime: bool


class TransitStopNearbyOut(BaseModel):
    stop_id: str
    name: str
    distance_meters: int
    departures: list[TransitDepartureOut]


class TransitNearbyResponse(BaseModel):
    city: str
    source: str
    mode: TransitMode
    disclaimer: str
    stops: list[TransitStopNearbyOut] = Field(default_factory=list)

File: frontend/packages/types/src/transit.ts
/** Grand Reims transit API types (WEB-MAP-02). */

export type TransitMode = "scheduled" | "realtime";

export type TransitDeparture = {
  route_short_name: string;
  route_type: string;
  headsign: string;
  scheduled_at: string;
  minutes?: number | null;
  realtime: boolean;
};

export type TransitStopNearby = {
  stop_id: string;
  name: string;
  distance_meters: number;
  departures: TransitDeparture[];
};

export type TransitNearbyParams = {
  lat: number;
  lon: number;
  city?: string;
  radius_meters?: number;
  limit?: number;
  max_minutes?: number;
};

export type TransitNearbyResponse = {
  city: string;
  source: string;
  mode: TransitMode;
  disclaimer: string;
  stops: TransitStopNearby[];
};
